smartsim/_core/mli/mli_schemas/response/response.capnp

```diff
@@ -0,0 +1,45 @@
+# BSD 2-Clause License
+
+# Copyright (c) 2021-2024, Hewlett Packard Enterprise
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+@0xa05dcb4444780705;
+
+using Tensors = import "../tensor/tensor.capnp";
+using ResponseAttributes = import "response_attributes/response_attributes.capnp";
+using DataRef = import "../data/data_references.capnp";
+
+struct Response {
+  status @0 :Int32;
```

Comment:
I see that the tests use a status==`200`, which appears to be a spin on the HTTP status code. I don't know that we should follow HTTP codes. 

- Would a status enum that we control the meaning of do?
- If we're stealing HTTP semantics, do we need `statusMessage`?
- If the `statusMessage` is really only useful when `status != 200`, then can we instead call it `errorMessage`?

Reply:
A status enum would totally do (I think)! And changing it to `errorMessage` makes sense to me as well. So ultimately the `Response` might look like:
```
struct Response {
  status @0 :StatusEnum; #tbd what these are
  errorInfo :union {
    errorMessage @1 :Text;
    none @2 :Void;
  }
  result :union {
    keys @3 :List(DataRef.TensorKey);
    data @4 :List(Tensors.Tensor);
  }
  customAttributes :union {
    torch @5 :ResponseAttributes.TorchResponseAttributes;
    tf @6 :ResponseAttributes.TensorFlowResponseAttributes;
    none @7 :Void;
  }
}
```

Or to avoid a union in `errorInfo` I suppose a successful response could have an empty string for `errorMessage`.